src/backend/joanie/lms_handler/backends/openedx.py

```diff
@@ -88,8 +90,7 @@ def set_enrollment(self, username, resource_link, active=True):
 
         if response.ok:
             data = json.loads(response.content)
-            if data["is_active"] == active:
-                return data
+            return data["is_active"]
```

Comment:
so I think I'm also checking that is_active is True. 
The change I'm making here is to not transpire what openedx returns  to the caller so that the backend always works the same whatever the LMS behind.
I return True if the user is enrolled and False if the user is unenrolled.

Reply:
You say this code was here for some reason but the tests didn't fail when I removed it so :man_shrugging: 
Then I started to understand your point and this had me thinking way too long ^^
In fact, if you request for an enrollment (is_active==True) and the LMS returns (is_active==False) then you want to mark the enrollment in joanie to `failed` right? Same for an unenrollment right? But then it means that we should have 2 separated fields on the Enrollment model : one to save the "desired" state of the enrollment, and one to save the status of our attempts to set it on the LMS? With this second field we can run retries and investigate synchronization issues?

Reply:
I refactored based on this discussion!